integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java

```diff
@@ -0,0 +1,668 @@
+
+    try {
+      logger.info("Creating domain");
+      createDomainAndVerify(domain, domainNamespace);
+
+      //check the desired completed, available and failed statuses
+      checkStatus("True", "False", "True");
+
+    } finally {
+      deleteDomainResource(domainUid, domainNamespace);
+    }
+  }
+
+
+  /**
+   * Test domain status failed condition with invalid node port. Verify the following conditions are
+   * generated: type: Completed, status: true type: Available, status: false type: Failed status: true
+   */
+  @Test
+  @DisplayName("Test domain status failed condition with invalid node port.")
+  void testInvalidNodePort() {
+    String image = MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG;
+
+    logger.info("Creating domain custom resource");
+    Domain domain = createDomainResource(domainUid, domainNamespace, adminSecretName,
+        OCIR_SECRET_NAME, encryptionSecretName, replicaCount, image);
+
+    AdminServer as = new AdminServer()
+        .serverStartState("RUNNING")
+        .adminService(new AdminService()
+            .addChannelsItem(new Channel()
+                .channelName("default")
+                .nodePort(19000)));
+    domain.getSpec().adminServer(as);
+
+    try {
+      logger.info("Creating domain");
+      createDomainAndVerify(domain, domainNamespace);
+
+      //check the desired completed, available and failed statuses
+      checkStatus("True", "False", "True");
+
+    } finally {
+      deleteDomainResource(domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * Test domain status failed condition with introspector failure. Verify the following conditions are
+   * generated: type: Completed, status: true type: Available, status: false type: Failed status: true
+   */
+  @Test
+  @DisplayName("Test domain status condition with introspector failure")
+  void testIntrospectorFailure() {
+    String image = MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG;
+
+    logger.info("Creating domain custom resource");
+    Domain domain = createDomainResource(domainUid, domainNamespace, adminSecretName,
+        OCIR_SECRET_NAME, encryptionSecretName, replicaCount, image);
+
+    domain.getSpec().configuration().introspectorJobActiveDeadlineSeconds(5L);
+
+    try {
+      logger.info("Creating domain");
+      createDomainAndVerify(domain, domainNamespace);
+
+      //check the desired completed, available and failed statuses
+      checkStatus("True", "False", "True");
+
+    } finally {
+      deleteDomainResource(domainUid, domainNamespace);
+    }
+  }
+
+
+  /**
+   * Test domain status condition with serverStartPolicy set to IF_NEEDED. Verify the following conditions are
+   * generated: type: Completed, status: true type: Available, status: true Verify no Failed type condition generated.
+   */
+  @Test
+  @DisplayName("Test domain status condition with serverStartPolicy set to IF_NEEDED")
+  void testMSBootFailureStatus() {
```

Comment:
The description does not mach the test name.

Reply:
changed